src/mono/System.Private.CoreLib/src/System/Threading/Monitor.Mono.cs

```diff
@@ -77,6 +82,12 @@ public static bool IsEntered(object obj)
         public static bool Wait(object obj, int millisecondsTimeout)
         {
             ArgumentNullException.ThrowIfNull(obj);
+#if FEATURE_WASM_THREADS
```

Comment:
Let's make helper method for this and localize the message.
The method should be "public" but suppressed same like `ThrowOnBlockingWaitOnJSInteropThread`.
So that we could use it in other places in the runtime.